### PR TITLE
DOC: Add caveat to to_stata with version=119

### LINKED ISSUE #37490: BUG: File saved with to_stata and version=119 cannot be opened with STATA 15.1 SE
- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import pandas as pd
df = pd.DataFrame({"A":[1]})
df.to_stata("a.dta", version=119)

```

#### Problem description

I have Stata/SE 15.1.

I run the above code, then try to open `a.dta` with my version of Stata.

But Stata doesn't open the file and instead gives the following error:

> dataset too large
>     This .dta file format was created by Stata/MP and has more variables than your Stata can handle.

Of course the file is not too large, since it contains only 1 observation and 1 variable.
It looks like pandas is saving the file in the "MP" version. 
I propose to save it in the SE version, or at least specify which sub-version of Stata the file is saved in in the documentation.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : db08276bc116c438d3fdee492026f8223584c477
python           : 3.8.5.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.18362
machine          : AMD64
processor        : Intel64 Family 6 Model 158 Stepping 10, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : Italian_Italy.1252

pandas           : 1.1.3
numpy            : 1.19.2
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.2.4
setuptools       : 50.3.0.post20201006
Cython           : 0.29.21
pytest           : 6.1.1
hypothesis       : None
sphinx           : 3.2.1
blosc            : None
feather          : None
xlsxwriter       : 1.3.7
lxml.etree       : 4.6.1
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.18.1
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : 1.3.2
fsspec           : 0.8.3
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.2
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 3.0.5
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.5.0
sqlalchemy       : 1.3.20
tables           : 3.6.1
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : 1.3.0
numba            : 0.51.2

</details>

### PR DESCRIPTION
**Pull request by hongshaoyang**
- [x] closes #37490
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Review by jbrockmendel — APPROVED** [review]
LGTM cc @bashtage

File: pandas/core/frame.py [hunk 1]
```diff
@@ -2129,7 +2129,7 @@ def to_stata(
             characters or fewer while versions 117 and later allow strings
             with lengths up to 2,000,000 characters. Versions 118 and 119
             support Unicode characters, and version 119 supports more than
-            32,767 variables.
```
Comment (bashtage) [inline pandas/core/frame.py:2132]: I wouldn't say "for now". Something along the lines of

"Version 119 should usually only be used when the number of variables exceeds the capacity of the dta version 118. Exporting smaller datasets to format 119 files may have unintended consequences, and, as of November 2020, Stata SE cannot read version 119 files. "

Comment (hongshaoyang) [reply pandas/core/frame.py:2132]: Thanks,

I copied your suggestion verbatim.

**Comment by bashtage** [comment]
LGTM.

**Comment by jreback** [comment]
thanks @hongshaoyang

**Pull request merged**